src/IonParserTextRaw.ts

```diff
@@ -263,6 +269,18 @@ export class ParserTextRaw {
     helpers[CH_MS] = this._read_value_helper_minus; // '-'
 
     this._read_value_helper_helpers = helpers;
+
+    helpers = {};
```

Comment:
Ok, I've restructured and simplified this whole helper initialization -- it's now factored as you suggest, and there's no more "patching", and there are no more magic ASCII numbers in the code